docs/coffeescript.js

```diff
@@ -0,0 +1,309 @@
+const coffeeSelector = document.getElementById("coffee-Selector");
+const teaSelector = document.getElementById("tea-Selector");
+const dessertSelector = document.getElementById("dessert-Selector");
+const grid = document.getElementById("menu-grid");
+const menuGrid = document.getElementById("menu-grid");
+const menuButton = document.getElementById("coffee-menu");
+const coffee_modal = document.getElementById(".modal");
+const navModal = document.querySelector(".nav-modal");
+let total;
+let lastValue;
+let products;
+let productsLength;
+
+// reading the material.JSON file
+async function fetchData() {
+  const response = await fetch("./material.json");
+  products = await response.json();
+  productsLength = products.length;
+  createMenuGrid(coffeeSelector.value);
+}
+
+fetchData();
+
+// script for creating grids
+function createMenuGrid(x) {
+  let gridContent = "";
+  let imageIndex = 1;
+  for (let i = 0; i < productsLength; i++) {
+    if (products[i].category === x) {
+      gridContent += ` 
+                    <div class="preview" id="previewCard${
+                      i + 1
+                    }" onclick="openModal('${x}',${imageIndex},${i})">
+                    <img
+                        src="./images/${x}-${imageIndex}.jpg"
+                        alt="Photo of a ${x}"
+                        class="preview image" />
+                    <h3 class="price-text coffee-name">${products[i].name}</h3>
+                    <p class="coffee-description">
+                    ${products[i].description}
+                    </p>
+                    <h3 class="price-text">$${products[i].price}</h3>
+                    </div>`;
+      imageIndex++;
+    }
+  }
+  return (grid.innerHTML = gridContent);
+}
+
+coffeeSelector.addEventListener("click", (e) => {
+  e.preventDefault();
+  createMenuGrid(coffeeSelector.value);
+});
+
+teaSelector.addEventListener("click", (e) => {
+  e.preventDefault();
+  createMenuGrid(teaSelector.value);
+});
+
+dessertSelector.addEventListener("click", (e) => {
+  e.preventDefault();
+  createMenuGrid(dessertSelector.value);
+});
+
+createMenuGrid(coffeeSelector.value);
+
+// script for the  coffee-modal calculations
+function checkSizeS(value) {
+  value = Math.abs(value);
+  total = Math.abs(total);
+  const sizeS = document.getElementById("size-s");
+  total -= lastValue;
+  if (sizeS.checked == 1) {
+    total = total + value;
+  }
+  lastValue = value;
+  document.querySelector(".total h3:last-child").innerText = `$${total.toFixed(
+    2
+  )}`;
+  return total;
+}
+
+function checkSizeM(value) {
+  value = Math.abs(value);
+  total = Math.abs(total);
+  const sizeM = document.getElementById("size-m");
+  total -= lastValue;
+  if (sizeM.checked == 1) {
```

Comment:
Always use type and value comparison - triple ===

Reply:
This 'If' is used to check whether if the respective element is selected or not. So it is just looking for a truthy value. If we use if triple === It won't work as we expect.

Reply:
Now I have changed the 1s to "true" and used triple equal signs.

Reply:
You also do not need to check if it equals true, you could do 
` (!!additive.checked  ? value : -value)`
that is not necessary to change now though. just for the future.